solc/CommandLineInterface.cpp

```diff
@@ -1194,6 +1194,8 @@ void CommandLineInterface::outputCompilationResults()
 		else
 			serr() << "Compiler run successful, no output requested." << endl;
 	}
+	else if (contracts.empty())
```

Comment:
Looks like we've got another problem here, which I've explained to the best of my abilities in the [original issue](https://github.com/ethereum/solidity/issues/4617#issuecomment-1302027761). We'll likely have to wait until Monday when @ekpyron and @cameel come back to provide more input - i.e. whether our approach here is just outright wrong and should be done elsewhere, or whether we should patch this with more fine grained checks. In the meantime, maybe @chriseth could chime in.

Reply:
We should simply put it inside the `if` above. I.e. only explain why there's no output if there's actually no output :)

Reply:
I've updated this to change the conditions under which it outputs the no contracts text. I haven't put it in the if statement above as suggested as m_hasOutput is always true when there is no contract to compile.

I put it in an else if, checked to ensure the output directory was empty so as not to get any non contract things that produce output and added the check for combinedJsonRequests so all tests pass. Assuming all the tests pass I will mark this as ready for review in the morning.

Reply:
> as m_hasOutput is always true when there is no contract to compile.

I don't think this is true. You can have both empty and non-empty output in that case, depending on which outputs you request.

Also, it's fine not to print this message when there is no output. The only reason we're printing that is that not printing anything may be confusing to the user. If there's some output already, this is not a concern.